src/workerd/api/sockets.h

```diff
@@ -167,7 +189,10 @@ class Socket: public jsg::Object {
   };
 
   void visitForGc(jsg::GcVisitor& visitor) {
-    visitor.visit(readable, writable);
+    visitor.visit(readable, writable, closedResolver,
+                  closedPromiseCopy, closedPromise,
+                  openedResolver, openedPromiseCopy,
+                  openedPromise);
```

Comment:
Entirely possible that there was a memory leak here previous with the promises not being traced.